### PR TITLE
Allow creating note by Ctrl clicking on non-legacy broken note link

### PR DESCRIPTION
**Pull request by fnkabit**
Feature #1698 
Video demo: https://www.youtube.com/watch?v=UFSM_hf7vh0

### CODE REVIEW

**Comment by fnkabit** [comment]
Should also add handling of case where sub-folders are not allowed.
In this case, would a warning message be enough (and automatically aborting operation) ?

**Comment by pbek** [comment]
Awesome, I'll test that today!

> Should also add handling of case where sub-folders are not allowed.
> In this case, would a warning message be enough (and automatically aborting operation) ?

Yes, that would be great. A warning message what note sub-folders are turned off for this note folder and aborting seems appropriate.

**Comment by pbek** [comment]
I now tested your PR, most cases already work! Thank you very much.

One case that didn't work was when I have in a sub-folder and add a link like `[Broken Link](../Broken-Note.md)`. I get the question if I want to create the note in the root folder but when I say `Yes` the note is created in the sub-folder instead of the root.

And it would be great if your mechanism would also work when you click on the note link in the preview.

**Comment by pbek** [comment]
I merged it now to let others test too. If you care, the clicking on the note link in the preview would also really be great.

**Comment by fnkabit** [comment]
Thank you Patrizio. I will try to reproduce the scenario that you mentioned.
In the next PR I will add the warning for when the subfolders are not allowed and also implement this same logic for note links in the preview.

**Comment by pbek** [comment]
Thank you very much.

> Thank you Patrizio. I will try to reproduce the scenario that you mentioned.

Like I said, I couldn't reproduce it any more. 😁

**Comment by pbek** [comment]
Two regressions so far. Happy days. 😆

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -6223,6 +6224,7 @@ void MainWindow::openLocalUrl(QString urlString) {
         Note note;
 
         if (isNoteFileUrl) {
+            ui->noteSubFolderTreeWidget->reset();
```
Comment (pbek) [inline src/mainwindow.cpp:6227]: @fnkabit, do you by change remember why you added this line? It causes #2128.

Comment (fnkabit) [reply src/mainwindow.cpp:6227]: @pbek Ah..sorry, don't remember.

Comment (pbek) [reply src/mainwindow.cpp:6227]: No worries, I removed the line for now. 😁

**Pull request merged**